### PR TITLE
Fix assert testing callmap return types

### PR DESCRIPTION
**Pull request by othercorey**
closes https://github.com/vimeo/psalm/issues/8675

This removes the assert that callmap return type is empty if PHP does not provide a return type.

Fixes a couple other errors I encountered when running callmap tests.

### CODE REVIEW

File: dictionaries/CallMap.php [hunk 1]
```diff
@@ -13683,7 +13683,7 @@
 'ssh2_exec' => ['resource|false', 'session'=>'resource', 'command'=>'string', 'pty='=>'string', 'env='=>'array', 'width='=>'int', 'height='=>'int', 'width_height_type='=>'int'],
 'ssh2_fetch_stream' => ['resource|false', 'channel'=>'resource', 'streamid'=>'int'],
 'ssh2_fingerprint' => ['string|false', 'session'=>'resource', 'flags='=>'int'],
-'ssh2_forward_accept' => ['resource|false', 'session'=>'resource'],
+'ssh2_forward_accept' => ['resource|false', 'listener'=>'resource'],
```
Comment (othercorey) [inline dictionaries/CallMap.php:13686]: The extension defined 2 optional parameters (host and port) that are unused in the code and undocumented in PHP docs but are reflected.

File: tests/Internal/Codebase/InternalCallMapHandlerTest.php [hunk 2]
```diff
@@ -282,6 +282,7 @@ class InternalCallMapHandlerTest extends TestCase
         'sqlsrv_prepare',
         'sqlsrv_query',
         'sqlsrv_server_info',
+        'ssh2_forward_accept',
```
Comment (othercorey) [inline tests/Internal/Codebase/InternalCallMapHandlerTest.php:285]: The extension defined 2 optional parameters (host and port) that are unused in the code and undocumented in PHP docs but are reflected.

File: tests/Internal/Codebase/InternalCallMapHandlerTest.php [hunk 3]
```diff
@@ -767,12 +719,12 @@ public function assertEntryReturnType(ReflectionFunction $function, string $entr
...
-        if ($expectedType === null) {
-            $this->assertSame('', $entryReturnType, 'CallMap entry has incorrect return type');
-            return;
-        }
 
-        $this->assertTypeValidity($expectedType, $entryReturnType, true, 'CallMap entry has incorrect return type');
+        if ($expectedType !== null) {
+            $this->assertTypeValidity($expectedType, $entryReturnType, true, 'CallMap entry has incorrect return type');
+        } else {
+            $this->assertNotEmpty($entryReturnType);
```
Comment (othercorey) [inline tests/Internal/Codebase/InternalCallMapHandlerTest.php:726]: I'm not sure if this is ever true or if the parse checks above catch this, but we should define a return type at least.

**Comment by orklah** [comment]
Thanks! That looks great!

**Pull request merged**